src/git.zsh

```diff
@@ -141,27 +141,35 @@ _fzf_complete_git() {
 }
 
 _fzf_complete_git_option_prefix() {
-    if [[ -z ${prefix:/--[^-]##*=*} ]]; then
+    if [[ -z ${prefix:/--[^-]##*=*} ]] || [[ ${prefix:0:2} =~ '-[^-]' ]]; then
         echo ${prefix/=*/=}
     fi
 }
+
 _fzf_complete_git_has_options() {
     local option
     local last_argument=$1
     local prefix=$2
     shift 2
 
     for option in ${(z)@}; do
-        if [[ ${#option} = 1 ]]; then
-            if [[ $last_argument =~ "^-[^-]*$option" ]]; then
+        local option_last_character=${option[-1]}
+        local option_without_last_question_mark=${option%\?}
+
+        if [[ ${#option_without_last_question_mark} = 1 ]]; then
+            if [[ $option_last_character != '?' ]] && [[ $last_argument =~ "^-[^-]*$option_without_last_question_mark" ]]; then
+                return 0
+            fi
+
+            if [[ $option_last_character == '?' ]] && [[ $prefix =~ "^-[^-]*$option_without_last_question_mark" ]]; then
```

Comment:
This line is wrong; think of the following case in which the commit message should be completed:
```bash
git commit -m**
```

Note: the current master branch does not handle this case as well.